sys/decl.go

```diff
@@ -292,14 +322,34 @@ func (t *PtrType) Align() uintptr {
 
 type StructType struct {
 	TypeCommon
-	Fields []Type
-	Varlen bool
-	padded bool
-	packed bool
-	align  uintptr
+	Fields         []Type
+	padded         bool
+	packed         bool
+	align          uintptr
+	varlen         bool
+	varlenAssigned bool
+}
+
+func (t *StructType) Varlen() bool {
+	if t.varlenAssigned {
```

Comment:
This is not thread-safe, and we don't necessary call t.Varlen from init function. So it can lead to data races.
call Varlen in the begining of addAlignment just to initialize it